src/ripple/app/misc/ValidatorList.h

```diff
@@ -547,12 +593,35 @@ class ValidatorList
 
     /** Return quorum for trusted validator set
 
-        @param trusted Number of trusted validator keys
+        @param unlSize Number of trusted validator keys
+
+        @param effectiveUnlSize Number of trusted validator keys that are not in
+        the NegativeUNL
 
-        @param seen Number of trusted validators that have signed
-        recently received validations */
+        @param seenSize Number of trusted validators that have signed
+        recently received validations
+    */
     std::size_t
-    calculateQuorum(std::size_t trusted, std::size_t seen);
+    calculateQuorum(
+        std::size_t unlSize,
+        std::size_t effectiveUnlSize,
+        std::size_t seenSize);
+
+    /**
+     * get the NodeIDs of Negative UNL validators
+     * @return the NodeIDs
+     */
+    hash_set<NodeID>
+    nUnlNodeIDs()
```

Comment:
Document that the lock needs to be held here.

Reply:
Right, I should. 
Now the function is removed.